barretenberg/cpp/src/barretenberg/vm2/tracegen/poseidon2_trace.cpp

```diff
@@ -291,8 +291,8 @@ void Poseidon2TraceBuilder::process_hash(
         auto num_perm_events = (input_size / 3) + static_cast<size_t>(input_size % 3 != 0);
         auto padded_size = 3 * ((event.inputs.size() + 2) / 3);
 
-        std::array<FF, 3> perm_input = { 0, 0, 0 };
         for (size_t i = 0; i < num_perm_events; i++) {
+            std::array<FF, 3> perm_input = { 0, 0, 0 };
```

Comment:
This was actually a little bug in `poseidon2_hash.pil` 😱. We weren't cleaning out the input array between permutation runs.